crates/noirc_evaluator/src/lib.rs

```diff
@@ -567,20 +626,31 @@ impl<'a> Evaluator<'a> {
                 let span = self.context.def_interner.expr_location(rhs);
                 let value =
                     self.evaluate_integer(env, rhs).map_err(|kind| kind.add_location(span))?;
-
-                env.store(variable_name, value);
+                env.store(variable_name, value, is_global);
             }
             Type::Array(..) => {
                 let rhs_poly = self.expression_to_object(env, rhs)?;
                 match rhs_poly {
                     Object::Array(arr) => {
-                        env.store(variable_name, Object::Array(arr));
+                        env.store(variable_name, Object::Array(arr), is_global);
                     }
                     _ => unimplemented!(
                         "The evaluator currently only supports arrays and constant integers!"
                     ),
                 };
             }
+            Type::PolymorphicInteger(is_const, typ_var) if is_const.is_const() => {
```

Comment:
This also entirely tied to evaluating the global const statements using `self.evaluate_statement`. This could either remain for further debugging of the evaluator with global consts (similarly to the global scope logic in `env` mentioned in my other comment), or be removed as we have a working example using the IR gen for compilation.